scripts/hid_configurator/configurator.py

```diff
@@ -533,23 +580,41 @@ def configurator():
-    subparsers = parser.add_subparsers(dest='command')
-    subparsers.required = True
-    parser_dfu = subparsers.add_parser('dfu', help='Run DFU')
-    parser_dfu.add_argument('dfu_image', type=str, help='Path to a DFU image')
-    parser_fwinfo = subparsers.add_parser('fwinfo', help='Obtain information about FW image')
-    parser_fwreboot = subparsers.add_parser('fwreboot', help='Request FW reboot')
-    parser_config = subparsers.add_parser('config', help='Write configuration option')
-    config_subparsers = parser_config.add_subparsers(dest='module')
-    config_subparsers.required = True
-    parser_config_sensor = config_subparsers.add_parser('sensor', help='Optical sensor options')
-    parser_config_sensor_subparsers = parser_config_sensor.add_subparsers(dest='option')
-    parser_config_sensor_subparsers.required = True
-    for opt_name in SENSOR_OPTIONS.keys():
-        parser_config_sensor_opt = parser_config_sensor_subparsers.add_parser(opt_name, help=SENSOR_OPTIONS[opt_name].help)
-        parser_config_sensor_opt.add_argument('value', help='value range: {} or "fetch" to fetch the value from device'.format(SENSOR_OPTIONS[opt_name].range))
+    sp_devices = parser.add_subparsers(dest='device_type')
+    sp_devices.required = True
+    for device_name in DEVICE.keys():
+        device_parser = sp_devices.add_parser(device_name)
+
+        sp_commands = device_parser.add_subparsers(dest='command')
+        sp_commands.required = True
+
+        parser_dfu = sp_commands.add_parser('dfu', help='Run DFU')
+        parser_dfu.add_argument('dfu_image', type=str, help='Path to a DFU image')
+
+        sp_commands.add_parser('fwinfo', help='Obtain information about FW image')
+        sp_commands.add_parser('fwreboot', help='Request FW reboot')
+
+        device_config = DEVICE[device_name]['config']
+        if device_config is not None:
+            assert(type(device_config) == dict)
+            parser_config = sp_commands.add_parser('config', help='Write configuration option')
+
+            sp_config = parser_config.add_subparsers(dest='module')
+            sp_config.required = True
+
+            for module_name in device_config:
+                module_config = device_config[module_name]
+                assert(type(module_config) == dict)
+                module_opts = module_config['options']
+                assert(type(module_opts) == dict)
+
+                parser_config_module = sp_config.add_parser(module_name, help='{} module options'.format(module_name))
+                sp_config_module = parser_config_module.add_subparsers(dest='option')
+                sp_config_module.required = True
+
+                for opt_name in module_opts:
+                    parser_config_module_opt = sp_config_module.add_parser(opt_name, help=module_opts[opt_name].help)
+                    parser_config_module_opt.add_argument('value', help='value range: {} or "fetch" to fetch the value from device'.format(module_opts[opt_name].range))
```

Comment:
Maybe add handling of improper values (curretly we get `ValueError: invalid literal for int() with base 10`).
Reproduced e.g. when we make typo in fetch

Reply:
Added